### DISCUSSION TITLE
can't install brew on m1 Big Sur 11.2.3

### DISCUSSION THREAD

**Discussion by bosserror404** [Homebrew/discussions#1182] [Everyday usage]
after hitting enter to install the below scrit

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

i just get the below script and brew refuses to install please can anyone help me out.... im not a programmer ... thanks you in advance.

curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused

**Comment by gromgit** [comment]
That just means the installer can't connect to GitHub.

Try running the command again and see if you get the same result. Might be a temporary issue.

**Reply by bosserror404** [reply]
i tried it again and again even after completely turning off the firewall still showing the below
"curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused"

**Reply by SMillerDev** [reply]
Can you open the link in your browser?

**Reply by bosserror404** [reply]
yes, i don't want to post all of it here but it,s a very long scrip ending like the one below.

 " cat <<EOS
    See ${tty_underline}https://docs.brew.sh/linux${tty_reset} for more information
- We recommend that you install GCC:
    brew install gcc

EOS
fi"

**Reply by SMillerDev** [reply]
how about `curl -vvv --head https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh`?

**Reply by bosserror404** [reply]
> how about `curl -vvv --head https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh`?

i retried to download it after running your script ... it started downloading then gave me this 

Downloading and installing Homebrew...
fatal: unable to access 'https://github.com/Homebrew/brew/': LibreSSL SSL_read: SSL_ERROR_SYSCALL, errno 60
Failed during: git fetch --force origin

**Reply by carlocab** [reply]
What's the output of
```
which -a git
which -a curl
```
and does the following command work?
```
git clone https://github.com/Homebrew/brew.git
```
(This should create a directory called `brew` in your current directory. Feel free to `rm -rf brew` if it does.)

**Reply by bosserror404** [reply]
> What's the output of
> 
> ```
> which -a git
> which -a curl
> ```
> 
> and does the following command work?
> 
> ```
> git clone https://github.com/Homebrew/brew.git
> ```
> 
> (This should create a directory called `brew` in your current directory. Feel free to `rm -rf brew` if it does.)

THIS IS WHAT I GOT

Cloning into 'brew'...
remote: Enumerating objects: 942, done.
remote: Counting objects: 100% (942/942), done.
remote: Compressing objects: 100% (398/398), done.
error: RPC failed; curl 56 LibreSSL SSL_read: error:1404C10B:SSL routines:ST_OK:wrong version number, errno 0
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed

**Reply by carlocab** [reply]
> What's the output of
> 
> ```
> which -a git
> which -a curl
> ```

**Reply by carlocab** [reply]
Sounds like you're facing a similar issue as #1185.

See https://github.com/Homebrew/discussions/discussions/1185#discussioncomment-567017.

**Reply by bosserror404** [reply]
> which -a git
> which -a curl

ITS OUTPUT IS 

/usr/bin/git
/usr/bin/curl

**Reply by carlocab** [reply]
Looks like a network issue. I can't help you much with that, as it's specific to your configuration. The links I posted in my last message might help.
